packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerCloudState.ts

```diff
@@ -0,0 +1,104 @@
+import { $TSContext, exitOnNextTick } from 'amplify-cli-core';
+import ora from 'ora';
+import { isMultiEnvLayer } from './layerHelpers';
+import { LayerVersionMetadata, PermissionEnum } from './layerParams';
+
+export class LayerCloudState {
+  private static instance: LayerCloudState;
+  private layerVersionsMetadata: LayerVersionMetadata[];
+  public latestVersionLogicalId: string;
+
+  static getInstance(): LayerCloudState {
+    if (!LayerCloudState.instance) {
+      LayerCloudState.instance = new LayerCloudState();
+    }
+    return LayerCloudState.instance;
+  }
+
+  private async loadLayerDataFromCloud(context: $TSContext, layerName: string): Promise<LayerVersionMetadata[]> {
+    const spinner = ora('Loading layer data from the cloud...').start();
+    let layerMetadata: LayerVersionMetadata[];
+    try {
+      const { envName }: { envName: string } = context.amplify.getEnvInfo();
+      const providerPlugin = await import(context.amplify.getProviderPlugins(context).awscloudformation);
+      const Lambda = await providerPlugin.getLambdaSdk(context);
+      const layerVersionList = await Lambda.listLayerVersions(isMultiEnvLayer(layerName) ? `${layerName}-${envName}` : layerName);
+      const Cfn = await providerPlugin.getCloudFormationSdk(context);
+      const stackList = await Cfn.listStackResources();
+      const layerStacks = stackList?.StackResourceSummaries?.filter(stack => stack.LogicalResourceId.includes(layerName));
+      let detailedLayerStack;
+
+      if (layerStacks?.length > 0) {
+        detailedLayerStack = (await Cfn.listStackResources(layerStacks[0].PhysicalResourceId)).StackResourceSummaries;
+      } else {
+        spinner.stop();
+        return [];
+      }
+      layerVersionList.forEach((layerVersion: LayerVersionMetadata) => {
+        let layerLogicalIdSuffix: string;
+        detailedLayerStack
+          .filter(stack => stack.ResourceType === 'AWS::Lambda::LayerVersion' && stack.PhysicalResourceId === layerVersion.LayerVersionArn)
+          .forEach(stack => {
+            layerVersion.LogicalName = stack.LogicalResourceId;
+            layerLogicalIdSuffix = stack.LogicalResourceId.replace('LambdaLayerVersion', '');
```

Comment:
`LambdaLayerVersion` should be a const as we construct the logical id at multiple places I assume?